src/TimeoutMigrationTool/ASQ/StagedDelayedMessageEntity.cs

```diff
@@ -0,0 +1,57 @@
+﻿namespace Particular.TimeoutMigrationTool.ASQ
+{
+    using Microsoft.Azure.Cosmos.Table;
+    using System;
+
+    /// <summary>
+    /// Represents a record in the native delays storage table which can be deferred message, saga timeouts, and delayed retries.
+    /// </summary>
+    public class StagedDelayedMessageEntity : TableEntity, ICanCalculateMySize
+    {
+        public string Destination { get; set; }
+        public byte[] Body { get; set; }
+        public string MessageId { get; set; }
+        public string Headers { get; set; }
+        public DateTimeOffset Time { get; set; }
```

Comment:
I seem to remember it is quite dangerous to store DateTimeOffset in table storage due to the automatic conversion that is happening https://stackoverflow.com/questions/28675770/why-doesnt-azure-table-storage-support-datetimeoffset

Reply:
Ok I guess this is already in use in ASQ so nothing we can do here